src/redux/hiddenTokens.ts

```diff
@@ -106,6 +110,10 @@ export const addHiddenToken = (tokenId: string) => (
   dispatch: Dispatch<HiddenTokensUpdateAction>,
   getState: AppGetState
 ) => {
+  const account = getState().wallets.selected!;
+
+  if (account.type === WalletTypes.readOnly) return;
```

Comment:
These checks are just insurance. The context menu option for "Hide" and the "Add to showcase" buttons have been hidden.